packages/lib/src/components/internal/FormFields/Select/Select.tsx

```diff
@@ -77,7 +77,8 @@ function Select(props: SelectProps) {
      * @param e - MouseEvent
      */
     const handleClickOutside = (e: MouseEvent) => {
-        if (!selectContainerRef.current.contains(e.target)) {
+        // use composedPath so it can also check when inside a web component
+        if (!e.composedPath().includes(selectContainerRef.current)) {
```

Comment:
[`composedPath`](https://caniuse.com/?search=composedPath) seems not to be available on some older browsers (e.g. IE11). We should add a fallback for those.